src/AggregateFunctions/DDSketchEncoding.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <vector>
+#include <stdexcept>
+
+namespace DB
+{
+class DDSketchEncoding
+{
+public:
+    static constexpr UInt8 numBitsForType = 2;
+    static constexpr UInt8 flagTypeMask = (1 << numBitsForType) - 1;
+    static constexpr UInt8 subFlagMask = ~flagTypeMask;
+
+    class Flag
+    {
+    public:
+        UInt8 byte;
+        Flag(UInt8 t, UInt8 s) : byte(t | s) { }
+        UInt8 Type() const { return byte & flagTypeMask; }
+        UInt8 SubFlag() const { return byte & subFlagMask; }
+    };
+
+    // FLAG TYPES
+    static constexpr UInt8 flagTypeSketchFeatures = 0b00;
+    static constexpr UInt8 FlagTypeIndexMapping = 0b10;
+    static constexpr UInt8 FlagTypePositiveStore = 0b01;
+    static constexpr UInt8 FlagTypeNegativeStore = 0b11;
+
+    // SKETCH FEATURES
+    const Flag FlagZeroCountVarFloat = Flag(flagTypeSketchFeatures, 1 << numBitsForType);
+
+    // INDEX MAPPING
+    const Flag FlagIndexMappingBaseLogarithmic = Flag(FlagTypeIndexMapping, 0 << numBitsForType);
+
+    // BINS
+    static constexpr UInt8 BinEncodingIndexDeltasAndCounts = 1 << numBitsForType;
+    static constexpr UInt8 BinEncodingIndexDeltas = 2 << numBitsForType;
```

Comment:
Similarly, this doesn't seem to be used anywhere as well.

Reply:
There are indeed some unused flags in the encoding.h but I left them deliberately to make the encoding consistent with the original sketch and be able to implement alternative store or mapping implementation in future.